x-pack/plugins/elastic_assistant/server/lib/langchain/tools/alert_counts/get_alert_counts_tool.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { SearchResponse } from '@elastic/elasticsearch/lib/api/types';
+import { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
+import { KibanaRequest } from '@kbn/core-http-server';
+import { DynamicTool, Tool } from 'langchain/tools';
+
+import { getAlertsCountQuery } from './get_alert_counts_query';
+import { requestHasRequiredAnonymizationParams } from '../../helpers';
+import type { RequestBody } from '../../types';
+
+export const ALERT_COUNTS_TOOL_DESCRIPTION =
+  'Call this for the counts of last 24 hours of open alerts in the environment, grouped by their severity';
+
+export const getAlertCountsTool = ({
+  alertsIndexPattern,
+  esClient,
+  replacements,
+  request,
+}: {
+  alertsIndexPattern?: string;
+  esClient: ElasticsearchClient;
+  replacements?: Record<string, string>;
+  request: KibanaRequest<unknown, unknown, RequestBody>;
+}): Tool | null => {
+  if (!requestHasRequiredAnonymizationParams(request) || alertsIndexPattern == null) {
+    return null;
+  }
+
+  return new DynamicTool({
+    name: 'alert-counts',
```

Comment:
Just a note that I'm trying to surface tool names in our traces, so we may want to standardize on some human readable/consistent format for tool names (esql tool is currently names `ESQLKnowledgeBaseTool`).

<img width="1091" alt="Screenshot_12_5_23__12_44" src="https://github.com/elastic/kibana/assets/2946766/11e97229-27ba-4e0b-b861-25bf4703ee1c">